Use overrides from @JsonProperty when it's provided

ana-vinogradova-camunda opened the pull request:
## Description

<!-- Describe the goal and purpose of this PR. -->
Use overrides from @JsonProperty when it's provided.

This is a bug fix for the following issue https://github.com/camunda-community-hub/spring-zeebe/issues/903 that was also present in the official SDK.
A fix for the same bug in the community project : https://github.com/camunda-community-hub/spring-zeebe/pull/934

What was done:
- Added field mapping from JsonProperty annotation
- Added test to verify the bugfix

## Checklist

<!--- Please delete options that are not relevant. Boxes should be checked by reviewer. -->
- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #21962

ana-vinogradova-camunda committed:
fix: do not ignore json field names

- Added field mapping from JsonProperty annotation
- Added test to verify the bugfix

mustafadagher commented on clients/spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/properties/PropertyBasedZeebeWorkerValueCustomizer.java:
```diff
@@ -117,10 +119,17 @@ private List<String> readVariablesAsTypeParameters(final MethodInfo methodInfo)
     return result;
   }
 
+  private String extractParameterName(final Field field) {
```
🔧 nitpicking: I'd have named this `extractFieldName`

ana-vinogradova-camunda replied:
Sounds better, yes 👍

mustafadagher reviewed (approved):
LGTM! 🚀 
only I minor suggestion below, but feel free to ignore it.

mustafadagher commented on clients/spring-boot-starter-camunda-sdk/src/test/java/io/camunda/zeebe/spring/client/properties/PropertyBasedZeebeWorkerValueCustomizerTest.java:
```diff
@@ -407,4 +426,9 @@ public void setVar4(final String var4) {
     }
   }
+
+  private static class PropertyAnnotatedClass {
```
final

ana-vinogradova-camunda committed:
refactor: refactoring the fix

- change method name to 'extractParameterName'
- make PropertyAnnotatedClass final

mustafadagher commented on clients/spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/properties/PropertyBasedZeebeWorkerValueCustomizer.java:
```diff
@@ -117,10 +119,17 @@ private List<String> readVariablesAsTypeParameters(final MethodInfo methodInfo)
 
+  private String extractFieldName(final Field field) {
+    if (field.isAnnotationPresent(JsonProperty.class)) {
+      return field.getAnnotation(JsonProperty.class).value();
```
❌ Actually, one final observation. The annotation could be present but the `value` could still be empty. resulting in a wrong field name.

We should make this check and add an extra test with `@JsonProperty` without a value

megglos commented:
@ana-vinogradova-camunda should we consider backporting this to 8.5/8.6?

ana-vinogradova-camunda commented:
> @ana-vinogradova-camunda should we consider backporting this to 8.5/8.6?

Yes. I will backport it.

ana-vinogradova-camunda committed:
fix: adding a check for non-empty annotation value

mustafadagher reviewed (approved):
Great work! Thanks a lot Ana!

backport-action commented:
Successfully created backport PR for `stable/8.6`:
- #23288
